### PR TITLE
Improve swagger instances, provide helper utilities...

### PR DESCRIPTION
**Pull request by sourabhxyz**
This PR:

* Adds few of utilities to work with Swagger.
* Adds `ToHttpApiData` instance for `GYMintingPolicyId`.
* Improves some of swagger instances with examples and descriptions and adds these instances for some of the types.
* Corrects `ToSchema` instance of `GYAssetClass`.
* Adds `signGYTx'` to support for `GYSomeSigningKey` like `signGYTxBody'`.

### CODE REVIEW

File: src/GeniusYield/Swagger/Utils.hs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+addSwaggerDescription :: (Functor f1, Functor f2, Swagger.HasSchema b1 a, Swagger.HasDescription a (Maybe b2)) => b2 -> f1 (f2 b1) -> f1 (f2 b1)
+addSwaggerDescription desc = mapped . mapped . Swagger.schema . Swagger.description ?~ desc
+
+-- | Utility function to add swagger example to a schema.
+addSwaggerExample :: (Functor f1, Functor f2, Swagger.HasSchema b1 a, Swagger.HasExample a (Maybe b2)) => b2 -> f1 (f2 b1) -> f1 (f2 b1)
+addSwaggerExample ex = mapped . mapped . Swagger.schema . Swagger.example ?~ ex
+
+-- | Drop the applied type symbol and convert camel case to snake case.
+dropSymbolAndCamelToSnake :: forall a. (KnownSymbol a) => String -> String
+dropSymbolAndCamelToSnake = camelTo2 '_' . drop (length $ symbolVal (Proxy @a))
```
Comment (4TT1L4) [inline src/GeniusYield/Swagger/Utils.hs:33]: nit: new line is missing at the end of the file:

```suggestion
dropSymbolAndCamelToSnake = camelTo2 '_' . drop (length $ symbolVal (Proxy @a))

```

File: src/GeniusYield/Types/Tx.hs [hunk 2]
```diff
@@ -108,6 +108,8 @@ instance Aeson.ToJSON GYTx where
...
...aae79402c8cccd5cd19b875007480008c848888888c014020c038d5d09aab9e500c23263533573802402602202001e01c01a01801601426aae7540104d55cf280189aab9e5002135573ca00226ea80048c8c8c8c8cccd5cd19b875001480088ccc888488ccc00401401000cdd69aba15004375a6ae85400cdd69aba135744a00646666ae68cdc3a80124000464244600400660106ae84d55cf280311931a99ab9c00b00c00a009008135573aa00626ae8940044d55cf280089baa001232323333573466e1d400520022321223001003375c6ae84d55cf280191999ab9a3370ea004900011909118010019bae357426aae7940108c98d4cd5ce00400480380300289aab9d5001137540022244464646666ae68cdc39aab9d5002480008cd5403cc018d5d0a80118029aba135744a004464c6a66ae7002002401c0184d55cf280089baa00149924103505431001200132001355008221122253350011350032200122133350052200230040023335530071200100500400132001355007222533500110022213500222330073330080020060010033200135500622225335001100222135002225335333573466e1c005200000d00c13330080070060031333008007335009123330010080030020060031122002122122330010040031122123300100300212200212200111232300100122330033002002001482c0252210853696c6c69636f6e003351223300248920975e4c7f8d7937f8102e500714feb3f014c8766fcf287a11c10c686154fcb27500480088848cc00400c00880050581840100d87980821a001f372a1a358a2b14f5f6" :: String)
```
Comment (4TT1L4) [inline src/GeniusYield/Types/Tx.hs:111]: Swagger `example` is deprecated. Now the use of "examples" is suggested instead.

**Comment by 4TT1L4** [comment]
FYI: GitHub added some warnings.

image

File: src/GeniusYield/Utils.hs [hunk 3]
```diff
@@ -48,4 +47,4 @@ serialiseToBech32WithPrefix prefix =
     Left e  -> error $ "serialiseToBech32WithPrefix: invalid prefix "
                     ++ show prefix
                     ++ ", " ++ show e
-    Right p -> serialiseToRawBytes >>> Bech32.dataPartFromBytes >>> Bech32.encodeLenient p
+    Right p -> serialiseToRawBytes >>> Bech32.dataPartFromBytes >>> Bech32.encodeLenient p
```
Comment (4TT1L4) [inline src/GeniusYield/Utils.hs:50]:
```suggestion
    Right p -> serialiseToRawBytes >>> Bech32.dataPartFromBytes >>> Bech32.encodeLenient p

```

File: src/GeniusYield/Types/Tx.hs (hunk 2 above)
Comment (sourabhxyz) [reply src/GeniusYield/Types/Tx.hs:111]: IIRC Swagger 2.0 doesn't support multiple examples.

**Comment by sourabhxyz** [comment]
Thanks @4TT1L4 for notifying about redundant import, I have removed it now along with some redundant libraries dependencies.
CC @ajuggler.

**Review by ajuggler — APPROVED** [review]
Reviewed.

**Pull request merged**